### ISSUE TITLE
Add case for holding at no fix

### ISSUE THREAD

**Issue by sam-irl** [openscope/openscope#1318]
Currently, issuing a command as such:
`AAL123 hold`
Yields:
`AAL123, unable to hold at unknown fix null`

Instead, this should readback something similar to
`AAL123, we weren't provided a fix to hold at`

**Comment by wrksx** [comment]
I wanted to fix this as my first issue, but I noticed the documentation says the command can be used like that: `AAL123 hold`

See (l. 186) https://github.com/openscope/openscope/blob/develop/documentation/aircraft-commands.md#hold

> _Syntax -_ `AAL123 hold [fixname] [left|right] [leg_time]min` or `AAL123 hold`

But it doesn't explain what should happen...
Should the documentation be fixed along this issue, or should such a command mean something?

**Comment by erikquinn** [comment]
@wrksx Correct, it should be able to be used as just `AAL123 hold`, in which case it should enter a present-position hold, right turns, 1 minute legs, as the documentation indicates. As far as I knew, this was working recently. The code should still exist, something probably just got disconnected-- or maybe it is trying to verify that _some parameters_ are provided, and it returns this message when none are found, rather than calling the appropriate methods.

Further investigation is needed to see why the appropriate logic is never reached. It should definitely still exist.

**Comment by wrksx** [comment]
@erikquinn Alright, I think I went through the documentation a bit too fast, it was indeed explained right there.
Does it make sense to keep this issue open and simply rephrase it?

**Comment by erikquinn** [comment]
Sure, or close and reopen a fresh one, either way.

**Comment by wrksx** [comment]
@erikquinn I didn't find the code that would allow to hold an arbirary position. I took a look at the code (and comments) to understand how it works, but I'm not sure if I'm taking the right direction with this one.

Currently, flying a holding pattern is handled by a specific logic (`_calculateTargetedGroundTrackLnav` in `AircraftModel`) applied to the current `WaypointModel` if it has the property `isHoldWaypoint` set to `true`.

I believe that to allow for an aircraft to hold it's current position it would require to create a waypoint at it's current position, but I'm not quite sure what to do with it **(1)**:
- add it to the current leg?
- create a new leg and insert it into the route?

Another issue is a `WaypointModel` can only get it's position through the `WaypointModel._initializePosition()` method, and that method rely on a list of determined fixes to grab the position.

I think  it's required to alter the WaypointModel and allow for the creation of an arbitrary waypoint to solve this issue **(2)**.

Please let me know what you guys think, specially regarding points marked **(1)** and **(2)**

**Comment by erikquinn** [comment]
Correct, we would have to do this with a new WaypointModel. The old way was kinda hacky and this will be much cleaner. As for where to insert the waypoint, I think it's probably as simple as "if the current leg is a procedural leg, insert the new waypoint at index 0 of the current leg. If the current leg is a direct leg, add a new direct leg with the new waypoint at index 0 of the LegCollection". Nothing jumps out at me in terms of pitfalls of that approach.

As for 2, I swore you could pass coordinates somehow. I'll check now.

**Comment by erikquinn** [comment]
Hm, I guess there actually isn't a way at the moment. So yes, we would have to add the ability to create waypoints from a positionmodel or coordinates. FYI, this new waypoint should be named `_PPOS`.

**Comment by cake-pie** [comment]
OP's request for readback to be adjusted [has been addressed](https://github.com/openscope/openscope/pull/1444/files#diff-cacfc1f363f58cbac43966995dd01e1e2ef1f8eca7b5cc760065adaf7666735eR306-R308) by #1444.

Restoring ability to enter standard holding pattern over present position is a less trival task.
